test/AsyncSpec.hs

```diff
@@ -21,22 +22,24 @@ spec = describe "async" $ do
       let message :: Member Trace r => Int -> String -> Sem r ()
           message n msg = trace $ mconcat
             [ show n, "> ", msg ]
-
+      ~[lock1, lock2] <- embed $
+        replicateM 2 newEmptyMVar
       a1 <- async $ do
           v <- get @String
           message 1 v
           put $ reverse v
 
-          embed $ threadDelay 1e5
+          embed $ putMVar lock1 ()
+          embed $ takeMVar lock2
           get >>= message 1
 
-          embed $ threadDelay 1e5
```

Comment:
I don't get what this `threadDelay` is for, so I simply removed it.

Reply:
This was so that the final `get` in this block would read the `pong` from the other thread.

Reply:
But this is already guaranteed (or should be, rather) from the previous `threadDelay`, isn't it?

Reply:
The logic I intended (perhaps not what I implemented) was this interleaving of threads:

thread1
thread2
thread1

Reply:
Yeah, that is what happens:
1. thread2 sleeps. thread1 gets, messages, puts, and sleeps
2. thread2 wakes, gets, messages, puts, and terminates
3. thread1 wakes, gets, messages, puts, and sleeps.
4. thread1 wakes, gets, and terminates.
5. main thread completes its `await` as thread1 has terminated, puts, and terminates.

The `threadDelay` I removed was the one in time slot 3, as it's seemingly redundant.